### PR TITLE
fix: Codespaces `corepack enable`

### PR DESCRIPTION
**Pull request by HowardBraham**
## **Description**

- #24930 broke Codespaces, this fixes it by adding `corepack enable`
- Had to use `yes | yarn --immutable` because it asks `Corepack is about to download... Do you want to continue? [Y/n]`
- Updated two `"yarn": "^4.0.2"` to `"yarn": "^4.2.2"`
- Properly deal with a 'Not Found' error in `yarn download-builds`
- disabled `git.autofetch`, as it was probably causing the `fatal: Cannot rebase onto multiple branches`

[Open in GitHub Codespaces](https://codespaces.new/MetaMask/metamask-extension/pull/25161?quickstart=1)

## **Manual testing steps**

1. Use the "Open in GitHub Codespaces" button, and make sure it works

## **Pre-merge author checklist**

- [x] I've followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Extension Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've completed the PR template to the best of my ability
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [x] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [x] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: .devcontainer/devcontainer.json [hunk 1]
```diff
@@ -56,5 +56,5 @@
 
   "runArgs": ["--shm-size=1g"],
 
-  "updateContentCommand": "sudo .devcontainer/install.sh && yarn --immutable && yarn tsx .devcontainer/setup-browsers.ts && echo 'export DISPLAY=:1' >> ~/.bashrc"
+  "updateContentCommand": "sudo .devcontainer/install.sh && corepack enable && yes | yarn --immutable && yarn tsx .devcontainer/setup-browsers.ts && echo 'export DISPLAY=:1' >> ~/.bashrc"
```
[the remark this answers is not in the document]
Comment (HowardBraham) [reply .devcontainer/devcontainer.json:59]: Ah quite interesting find!

**Review by legobeat — DISMISSED** [review]

**Review by plasmacorral — DISMISSED** [review]
Works, LGTM.

File: .devcontainer/devcontainer.json [hunk 2]
```diff
@@ -48,7 +48,7 @@
     }
   },
 
-  "postAttachCommand": "/usr/local/share/desktop-init.sh && git pull --rebase; yarn download-builds",
```
[the remark this answers is not in the document]
Comment (HowardBraham) [reply .devcontainer/devcontainer.json:51]: It was producing a weird error that Mike noticed where it says `fatal: Cannot rebase onto multiple branches.`

Also install.sh sets `git config --global pull.rebase true` anyway.

**Review by plasmacorral — DISMISSED** [review]
Error resolved, LGTM.

**Review by hjetpoluru — APPROVED** [review]
LGTM

**Review by legobeat — APPROVED** [review]

**Review by DDDDDanica — APPROVED** [review]

**Review by danjm — APPROVED** [review]

**Pull request merged**